### ISSUE TITLE
Assert prevents mixed_precision?

### ISSUE THREAD

**Issue by mattmisk** [lucidrains/imagen-pytorch#357]
Line 2661 of imagen_pytorch:

`assert images.dtype == torch.float, f'images tensor needs to be floats but {images.dtype} dtype found instead'`

This assert triggers when using mixed_precision = True.  Am I doing something wrong, or is this a bug? Everything works fine if I comment the line out.

**Comment by lucidrains** [comment]
@mattmisk oh nice catch! put in a [quick fix](https://github.com/lucidrains/imagen-pytorch/commit/f9fb8beb69eea15454153226f3a0d8323152d66b)

**Comment by lucidrains** [comment]
@mattmisk generated images look ok?

**Comment by mattmisk** [comment]
> @mattmisk generated images look ok?

Yes, but my application is unique and still in development (dataset acquisition, processing, etc.). It's on my list to do a head-to-head comparison with and without mixed precision.
